### DISCUSSION TITLE
Duda del Ejercicio 2 de la Kata 4

### DISCUSSION THREAD

**Discussion by Juan-2003** [LaunchX-InnovaccionVirtual/CursoIntroPython#115] [Q&A]
Buenas chicos / chicas. Al momento de hacer ir realizando los ejercicios realmente no entendí esta solución. Se para que sirven los "for", pero aquí realmente si me perdí. ¿Alguien que me pueda explicar?

![duda1](https://user-images.githubusercontent.com/99064991/153535646-eed4cde4-0a2a-4705-afb8-8ea35548da28.PNG)
![duda](https://user-images.githubusercontent.com/99064991/153535647-d4df882a-c823-4634-95b8-e79515c117b8.PNG)

**Comment by AlexString** [comment]
Lo importante es ir entendiendo cada sentencia, espero que un par de comentarios te ayuden a comprender mejor:
```python
for sentence in text_parts: # por cada sentencia en partes del texto
    for key_word in key_words: # por cada palabra clave en palabras clave
        if key_word in sentence: # si la palabra clave se encuentra en la sentencia:
            print(sentence.replace(' C', ' Celsius')) # Donde encuentres el caracter 'C' reemplazalo por 'Celsius' además, imrpimelo
            break
```
Donde:
- `sentence` es un objeto de tipo `string` y los objetos de tipo `string` tienen métodos, especificamente en este ejemplo se usa el método `.replace()`
- Toma en cuenta que al mencionar **por cada** en mis comentarios quiero darte a entender que "por cada elemento, se realiza una iteración del ciclo/bucle"

Practicamente el segundo bucle anidado hace el mismo **recorrido** (lo llamamos así) pero este solamente imprime sin reemplazar.

Si te quedó la curiosidad de saber más puedes buscar sobre:
- Bucles en python
- Programación Orientada a Objetos en Python (pero una vez que aprendes la teoria de la POO entiendes la POO en los demás lenguajes, es un paradigma que se utiliza mucho).

**Comment by doguedogue** [comment]
ante la duda usa prints en todo momento en tus ciclos, revisa que tienen, que resulta de las operaciones
```
for oracion in text.split('.'):
    print (oracion)
    for palabra in palabras_clave:
        print(palabra)
```
